### PR TITLE
Add endpoint find

### PR DESCRIPTION
**Pull request by ekaterinaryzhenkova**
Add endpoint to get current configuration of all services from cache.

### CODE REVIEW

File: src/AdminService.Business/Commands/FindAdminCommand.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      _configrepository = configrepository;
+      _configmapper = configmapper;
+      _httpContextAccessor = httpContextAccessor;
+    }
+
+    public async Task<FindResultResponse<ConfigurationServicesInfo>> ExecuteAsync(FindAdminFilter filter)
+    {
+      if(!_baseFindValidator.ValidateCustom(filter, out List<string> errors))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
```
Comment (Cheephot) [inline src/AdminService.Business/Commands/FindAdminCommand.cs]: use ResponseCreator (Kernel)

File: src/AdminService.Business/Commands/FindAdminCommand.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        return new()
+        {
+          Status = OperationResultStatusType.Failed,
+          Errors = errors
+        };
+      }
+
+      List<DbServiceConfiguration> dbConfig = null; 
+      FindResultResponse<ConfigurationServicesInfo> response = new();
+    //  response.Body = new();
```
Comment (Cheephot) [inline src/AdminService.Business/Commands/FindAdminCommand.cs]: remove

File: src/AdminService.Business/Commands/FindAdminCommand.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+      (List<DbServiceConfiguration> dbConfig, int TotalCount) findServicesResponse =
+        await _configrepository.FindAsync(filter);
+
+      dbConfig = findServicesResponse.dbConfig;
```
Comment (Cheephot) [inline src/AdminService.Business/Commands/FindAdminCommand.cs]: this in an extra variable, use findServicesResponse.dbConfig in 60 line

File: src/AdminService.Data/ServiceConfigurationRepository.cs [hunk 4]
```diff
@@ -1,8 +1,33 @@
...
+using LT.DigitalOffice.AdminService.Data.Provider;
+using LT.DigitalOffice.AdminService.Models.Db;
+using LT.DigitalOffice.AdminService.Models.Dto.Requests.Filtres;
+using Microsoft.EntityFrameworkCore;
+
+namespace LT.DigitalOffice.AdminService.Data
+{
+  public class ServiceConfigurationRepository : IServiceConfigurationRepository
+  {
+    private readonly IDataProvider _provider;
```
Comment (Cheephot) [inline src/AdminService.Data/ServiceConfigurationRepository.cs:14]: add line

File: src/AdminService.Data/ServiceConfigurationRepository.cs [hunk 5]
```diff
@@ -1,8 +1,33 @@
...
+    public ServiceConfigurationRepository(IDataProvider provider)
+    {
+      _provider = provider;
+    }
```
Comment (Cheephot) [inline src/AdminService.Data/ServiceConfigurationRepository.cs:19]: add line

File: src/AdminService.Data/ServiceConfigurationRepository.cs [hunk 6]
```diff
@@ -1,8 +1,33 @@
...
+    public async Task<(List<DbServiceConfiguration> dbconfig, int totalCount)> FindAsync(FindAdminFilter filter)
+    {
+      if(filter == null)
```
Comment (Cheephot) [inline src/AdminService.Data/ServiceConfigurationRepository.cs]: if "space" (filter is null)

File: src/AdminService.Mappers/ConfigurationServicesInfoMapper.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+using LT.DigitalOffice.AdminService.Models.Db;
+using LT.DigitalOffice.AdminService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.AdminService.Mappers
+{
+  internal class ConfigurationServicesInfoMapper : IConfigurationServicesInfoMapper
+  {
+    public ConfigurationServicesInfo Map(DbServiceConfiguration dbconfig)
+    {
+      if(dbconfig == null)
```
Comment (Cheephot) [inline src/AdminService.Mappers/ConfigurationServicesInfoMapper.cs]: same

File: src/AdminService.Models.Dto/Models/ConfigurationServicesInfo.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.AdminService.Models.Dto.Models
+{
+  public class ConfigurationServicesInfo
+  {
+    public Guid Id { get; set; }
+    public string ServiceName { get; set; }
+    public bool IsActive { get; set; }
+
```
Comment (Cheephot) [inline src/AdminService.Models.Dto/Models/ConfigurationServicesInfo.cs]: remove line

File: src/AdminService.Business/Commands/FindAdminCommand.cs [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+using LT.DigitalOffice.Kernel.FluentValidationExtensions;
+using LT.DigitalOffice.Kernel.Helpers;
+using LT.DigitalOffice.Kernel.Helpers.Interfaces;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.Kernel.Validators.Interfaces;
+using Microsoft.AspNetCore.Http;
+
+namespace LT.DigitalOffice.AdminService.Business.Commands
+{
+  public class FindAdminCommand : IFindAdminCommand
```
Comment (MariyaSin) [inline src/AdminService.Business/Commands/FindAdminCommand.cs]: rename to FindServiceConfigurationCommand

File: src/AdminService.Business/Commands/FindAdminCommand.cs [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    {
+      if(!_baseFindValidator.ValidateCustom(filter, out List<string> errors))
+      {
+        return _responseCreator.CreateFailureFindResponse<ConfigurationServicesInfo>(HttpStatusCode.BadRequest, errors);
+      };     
+
+      FindResultResponse<ConfigurationServicesInfo> response = new();
+      response.Body = new();
+
+      (List<DbServiceConfiguration> dbConfig, int totalCount) findServicesResponse =
```
Comment (MariyaSin) [inline src/AdminService.Business/Commands/FindAdminCommand.cs]: you can do not crete  findServicesResponse vdriable: (List<DbServiceConfiguration> dbConfig, int totalCount) = ...

File: src/AdminService.Business/Commands/FindAdminCommand.cs [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+        await _configrepository.FindAsync(filter);
+
+      List<DbServiceConfiguration> dbConfig = findServicesResponse.dbConfig;
```
Comment (MariyaSin) [inline src/AdminService.Business/Commands/FindAdminCommand.cs]: no need create the variable

File: src/AdminService.Business/Commands/FindAdminCommand.cs [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+      response.TotalCount = findServicesResponse.totalCount;
+      
+      response.Body.AddRange(dbConfig.Select((dbConfig) => _configmapper.Map(dbConfig)));
+
+      response.Status = response.Errors.Any()
```
Comment (MariyaSin) [inline src/AdminService.Business/Commands/FindAdminCommand.cs]: any way you do not have errors here

File: src/AdminService.Data.Provider/IDataProvider.cs [hunk 13]
```diff
@@ -9,6 +9,7 @@ namespace LT.DigitalOffice.AdminService.Data.Provider
   [AutoInject(InjectType.Scoped)]
   public interface IDataProvider : IBaseDataProvider
   {
-    IQueryable<DbServiceConfiguration> ServicesConfigurations { get; }
+    IQueryable<DbServiceConfiguration> ServicesConfigurationsQueryable { get; }
+
```
Comment (MariyaSin) [inline src/AdminService.Data.Provider/IDataProvider.cs]: remove empty line?

File: src/AdminService.Models.Dto/Models/ConfigurationServicesInfo.cs [hunk 14]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace LT.DigitalOffice.AdminService.Models.Dto.Models
+{
+  public class ConfigurationServicesInfo
```
Comment (MariyaSin) [inline src/AdminService.Models.Dto/Models/ConfigurationServicesInfo.cs]: use record instead class.
Rename to ServiceConfigurationInfo

File: src/AdminService.Models.Dto/Requests/Filtres/FindAdminFilter.cs [hunk 15]
```diff
@@ -0,0 +1,8 @@
+﻿using LT.DigitalOffice.Kernel.Requests;
+
+namespace LT.DigitalOffice.AdminService.Models.Dto.Requests.Filtres
+{
+  public record FindAdminFilter : BaseFindFilter
```
Comment (MariyaSin) [inline src/AdminService.Models.Dto/Requests/Filtres/FindAdminFilter.cs]: you can remove it and only use BaseFindFilter

File: src/AdminService.Models.Dto/Responses/FindResponse.cs [hunk 16]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using LT.DigitalOffice.AdminService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.AdminService.Models.Dto.Responses
+{
+  public record FindResponse
```
Comment (MariyaSin) [inline src/AdminService.Models.Dto/Responses/FindResponse.cs]: you use FindResultResponse, remove the class

**Review by Cheephot — APPROVED** [review]

File: src/AdminService.Business/Commands/FindServiceConfigurationCommand.cs [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+using LT.DigitalOffice.Kernel.Requests;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.Kernel.Validators.Interfaces;
+
+namespace LT.DigitalOffice.AdminService.Business.Commands
+{
+  public class FindServiceConfigurationCommand : IFindServiceConfigurationCommand
+  {
+    private readonly IBaseFindFilterValidator _baseFindValidator;
+    private readonly IServiceConfigurationRepository _configrepository;
```
Comment (MariyaSin) [inline src/AdminService.Business/Commands/FindServiceConfigurationCommand.cs]: please write the word configuration in full

Comment (MariyaSin) [reply src/AdminService.Business/Commands/FindServiceConfigurationCommand.cs]: use camel case

File: src/AdminService.Business/Commands/FindServiceConfigurationCommand.cs [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+    private readonly IConfigurationServicesInfoMapper _configmapper;
```
Comment (MariyaSin) [inline src/AdminService.Business/Commands/FindServiceConfigurationCommand.cs]: same

File: src/AdminService.Business/Commands/FindServiceConfigurationCommand.cs [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+    private readonly IResponseCreater _responseCreator;
+
+    public FindServiceConfigurationCommand(
+      IBaseFindFilterValidator baseFindValidator,
+      IServiceConfigurationRepository configrepository,
```
Comment (MariyaSin) [inline src/AdminService.Business/Commands/FindServiceConfigurationCommand.cs]: same

File: src/AdminService.Business/Commands/FindServiceConfigurationCommand.cs [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+      IConfigurationServicesInfoMapper configmapper,
```
Comment (MariyaSin) [inline src/AdminService.Business/Commands/FindServiceConfigurationCommand.cs]: same

File: src/AdminService.Data/Interfaces/IServiceConfigurationRepository.cs [hunk 21]
```diff
@@ -1,6 +1,14 @@
...
+using LT.DigitalOffice.AdminService.Models.Db;
+using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.Kernel.Requests;
+
+namespace LT.DigitalOffice.AdminService.Data.Interfaces
+{
+  [AutoInject]
+  public interface IServiceConfigurationRepository
+  {
+    Task<(List<DbServiceConfiguration> dbconfig, int totalCount)> FindAsync(BaseFindFilter filter);
```
Comment (MariyaSin) [inline src/AdminService.Data/Interfaces/IServiceConfigurationRepository.cs]: dbServicesConfigurations

File: src/AdminService.Business/Commands/FindServiceConfigurationCommand.cs [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+    {
+      if (!_baseFindValidator.ValidateCustom(filter, out List<string> errors))
+      {
+        return _responseCreator.CreateFailureFindResponse<ServiceConfigurationInfo>(HttpStatusCode.BadRequest, errors);
+      };
+
+      FindResultResponse<ServiceConfigurationInfo> response = new();
+      response.Body = new();
+
+      (List<DbServiceConfiguration> dbConfig, int totalCount) =
```
Comment (MariyaSin) [inline src/AdminService.Business/Commands/FindServiceConfigurationCommand.cs]: dbServicesConfigurations

File: src/AdminService.Data/ServiceConfigurationRepository.cs [hunk 23]
```diff
@@ -1,8 +1,34 @@
...
+{
+  public class ServiceConfigurationRepository : IServiceConfigurationRepository
+  {
+    private readonly IDataProvider _provider;
+
+    public ServiceConfigurationRepository(IDataProvider provider)
+    {
+      _provider = provider;
+    }
+    public async Task<(List<DbServiceConfiguration> dbconfig, int totalCount)> FindAsync(BaseFindFilter filter)
```
Comment (MariyaSin) [inline src/AdminService.Data/ServiceConfigurationRepository.cs]: dbServicesConfigurations

File: src/AdminService.Data/ServiceConfigurationRepository.cs [hunk 24]
```diff
@@ -1,8 +1,34 @@
...
+    {
+      if (filter is null)
+      {
+        return (null, default);
+      }
+
+      IQueryable<DbServiceConfiguration> dbconfig = _provider.ServicesConfigurationsQueryable;
```
Comment (MariyaSin) [inline src/AdminService.Data/ServiceConfigurationRepository.cs]: same

File: src/AdminService.Mappers/ConfigurationServicesInfoMapper.cs [hunk 25]
```diff
@@ -0,0 +1,24 @@
+﻿using LT.DigitalOffice.AdminService.Mappers.Interfaces;
+using LT.DigitalOffice.AdminService.Models.Db;
+using LT.DigitalOffice.AdminService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.AdminService.Mappers
+{
+  public class ConfigurationServicesInfoMapper : IConfigurationServicesInfoMapper
```
Comment (MariyaSin) [inline src/AdminService.Mappers/ConfigurationServicesInfoMapper.cs]: ServiceConfigurationInfoMapper

File: src/AdminService.Mappers/ConfigurationServicesInfoMapper.cs [hunk 26]
```diff
@@ -0,0 +1,24 @@
...
+using LT.DigitalOffice.AdminService.Models.Db;
+using LT.DigitalOffice.AdminService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.AdminService.Mappers
+{
+  public class ConfigurationServicesInfoMapper : IConfigurationServicesInfoMapper
+  {
+    public ServiceConfigurationInfo Map(DbServiceConfiguration dbconfig)
+    {
+      if(dbconfig is null)
```
Comment (MariyaSin) [inline src/AdminService.Mappers/ConfigurationServicesInfoMapper.cs]: same

File: src/AdminService.Mappers/Interfaces/IConfigurationServicesInfoMapper.cs [hunk 27]
```diff
@@ -0,0 +1,12 @@
+﻿using LT.DigitalOffice.AdminService.Models.Db;
+using LT.DigitalOffice.AdminService.Models.Dto.Models;
+using LT.DigitalOffice.Kernel.Attributes;
+
+namespace LT.DigitalOffice.AdminService.Mappers.Interfaces
+{
+  [AutoInject]
+  public interface IConfigurationServicesInfoMapper
```
Comment (MariyaSin) [inline src/AdminService.Mappers/Interfaces/IConfigurationServicesInfoMapper.cs]: IServiceConfigurationInfoMapper

**Review by grinakate — APPROVED** [review]

File: src/AdminService.Mappers/ServiceConfigurationInfoMapper.cs [hunk 28]
```diff
@@ -0,0 +1,24 @@
...
+using LT.DigitalOffice.AdminService.Models.Db;
+using LT.DigitalOffice.AdminService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.AdminService.Mappers
+{
+  public class ServiceConfigurationInfoMapper : IServiceConfigurationInfoMapper
+  {
+    public ServiceConfigurationInfo Map(DbServiceConfiguration dbServicesConfigurations)
+    {
+      if(dbServicesConfigurations is null)
```
Comment (VladSpitsin) [inline src/AdminService.Mappers/ServiceConfigurationInfoMapper.cs]: add space after "if"

File: src/AdminService/Startup.cs [hunk 29]
```diff
@@ -1,265 +1,267 @@
...
+
+      _rabbitMqConfig = Configuration
+        .GetSection(BaseRabbitMqConfig.SectionName)
+        .Get<RabbitMqConfig>();
+
+      _serviceInfoConfig = Configuration
+        .GetSection(BaseServiceInfoConfig.SectionName)
+        .Get<BaseServiceInfoConfig>();
+
+      Version = "1.0.0.0";
```
Comment (VladSpitsin) [inline src/AdminService/Startup.cs:51]: maybe you need to change version? consult with Mariya

File: src/AdminService/Startup.cs [hunk 30]
```diff
@@ -1,265 +1,267 @@
...
+      using var serviceScope = app.ApplicationServices
+        .GetRequiredService<IServiceScopeFactory>()
+        .CreateScope();
+
+      using var context = serviceScope.ServiceProvider.GetService<AdminServiceDbContext>();
+
+      context.Database.Migrate();
+    }
+
+    private string HidePassord(string line)
```
Comment (VladSpitsin) [inline src/AdminService/Startup.cs]: remove this method and use HidePassword from Kernel.Helpers like Spartak and Mariya said

File: src/AdminService.Models.Dto/Configuration/RabbitMqConfig.cs [hunk 31]
```diff
@@ -1,15 +1,15 @@
...
-
+﻿using LT.DigitalOffice.Kernel.BrokerSupport.Attributes;
+using LT.DigitalOffice.Kernel.BrokerSupport.Configurations;
+using LT.DigitalOffice.Models.Broker.Requests.User;
+
+namespace LT.DigitalOffice.AdminService.Models.Dto.Configuration
+{
+  public class RabbitMqConfig : BaseRabbitMqConfig
+  {
+    public string UpdateSmtpCredentialsEndpoint { get; set; }
```
Comment (MariyaSin) [inline src/AdminService.Models.Dto/Configuration/RabbitMqConfig.cs]: [AuthoInjectRequest(typeof(ICreateSmtpCredentialsRequest))]
public string CreateSmtpCredentialsEndpoint { get; set; }

File: src/AdminService/appsettings.json [hunk 32]
```diff
@@ -1,55 +1,56 @@
...
+
+    //AdminService
+    "GetSmtpCredentialsEndpoint": "AdminService_GetSmtpCredentials",
+
+    // UserService
+    "CheckUserIsAdminEndpoint": "UserService_GetUserAdminStatus",
+    "CreateAdminEndpoint": "UserService_CreateAdmin",
+
+    // EmailService
+    "UpdateSmtpCredentialsEndpoint": "EmailService_UpdateSmtpCredentials"
```
Comment (MariyaSin) [inline src/AdminService/appsettings.json]: "CreateSmtpCredentialsEndpoint" : "EmailService_CreateSmtpCredentials"

**Review by MariyaSin — APPROVED** [review]

**Review by VladSpitsin — APPROVED** [review]

**Pull request merged**